app/models/story.rb

```diff
@@ -30,6 +32,7 @@ class Story < ActiveRecord::Base
   has_many :notifications, as: :source, dependent: :destroy
 
   validates :user, :story_type, presence: true
+  validates :group, presence: true, if: Proc.new {|s| s.group_id.present? }
```

Comment:
I am so confused, doesn't this literally mean "if group is present, check that group is present"?